File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode a, ListNode b) {

        if (a == null) {
            return b;
        }
        
        if (b == null) {
            return a;
        }

        var result = (ListNode) null;

        if (a.val < b.val) {
            result = new ListNode(a.val);
            a = a.next;
        } else {
            result = new ListNode(b.val);
            b = b.next;
        }

        var tail = result;

        while (a != null || b != null) {
            if (a == null) {
                tail.next = new ListNode(b.val);
                b = b.next;
            } else if (b == null) {
                tail.next = new ListNode(a.val);
                a = a.next;
            } else if (a.val > b.val) {
                tail.next = new ListNode(b.val);
                b = b.next;
            } else {
                tail.next = new ListNode(a.val);
                a = a.next;
            }
            tail = tail.next;
        }

        return result;
    }
}

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        dummy = ListNode()
        tail = dummy
        queue = deque()
        i = 0
        current = head
        while current:
            queue.append(current)
            i += 1
            current = current.next
            if i % k == 0:
                while len(queue) > 0:
                    tail.next = queue.pop()
                    tail = tail.next
                tail.next = None
        while len(queue) > 0:
            tail.next = queue.popleft()
            tail = tail.next
        tail.next = None
        return dummy.next
